dev/source/docs/ros2-over-ethernet.rst

```diff
@@ -0,0 +1,94 @@
+.. _ros2-over-ethernet:
+
+===================
+ROS 2 over Ethernet
+===================
+
+The following tutorial explains how to connect a flight controller to a companion computer using ethernet for DDS communication.
+
+Ensure you have ROS 2 :ref:`installed <ros2>` and have run :ref:`SITL <ros2-sitl>` successfully before attempting this page.
+
+Additionally, make sure you understand the :ref:`basics of networking in ArduPilot. <common-network>`
+
+==========
+Motivation
+==========
+
+Starting in ArduPilot 4.5, ethernet is now supported as a new communication interface.
+Compared to connecting an autopilot over serial, ethernet has the following advantages:
+
+* Ethernet is more immune to noise because it uses twisted pair wiring
+* Ethernet can run multiple protocols over the same protocol (unless you use PPP)
+* Ethernet is easier to debug with tools such as ``tcpdump`` or ``Wireshark``
+* Ethernet uses standard cabling, so it's more difficult to mix up TX and RX
+
+===============
+Physical Equipment
+===============
+
+The following equipment is required to complete this tutorial:
+
+* A computer that can run the MicroROS Agent and the ROS 2 CLI
+* An autopilot with ethernet support
+
+=============================================
+Necessary parameters for static configuration
+=============================================
+
+This list of parameters is given as an example.
+If you had the following static IP addresses:
+
+* An autopilot has the IP address ``192.168.1.6``
+* A Computer running the MicroROS agent has the IP address ``192.168.1.5``
+* The MicroROS agent is running on port ``2019``
+
+Then, you would configure all of the below parameters.
+
+
+.. list-table::
+   :widths: 50 50
+   :header-rows: 1
+
+   * - Parameter Name
+     - Value
+   * - DDS_ENABLE
+     - 1
+   * - NET_ENABLED
+     - 1
+   * - NET_DHCP
+     - 0
+   * - NET_IP_ADDR0
```

Comment:
most params names are wrong...and each needs a hotlink...many net params are missing like client/server type and port

Reply:
I only found two that were wrong. They are all hyperlinked now. Note: I didn't know if you generally want to add user parameters if they are rarely changed.

Any others that are missing?

Reply:
actually you correted 6 not two...and any param that needs to be a certain value should be listed in case someone messes them up. they then have a reference to restore operation...unless obvious like net_enable

Reply:
How do I fix this warning: `/home/ryan/Dev/ros2_ws/src/ardupilot_wiki/dev/source/docs/ros2-over-ethernet.rst:64: WARNING: undefined label: 'dds_port'`

The param is `DDS_PORT`.

Reply:
I found a bug from the DDS implementation and proposed a fix here. Hold off on merging this PR till this is fixed:
https://github.com/ArduPilot/ardupilot/pull/26577

Reply:
Upstream PR merged. Can you re-run the build?

Reply:
takes over a day to get changed in the inventory....and....you did not change the param name here so it will fail...still needs a squash

Reply:
Done and done. There's now two commits, one per subsystem.

Reply:
```
commit 22ef1a3a431513858ef97f150152446429e0ceed (HEAD -> add-ros2-ethernet-docs, origin/add-ros2-ethernet-docs)
Author: Ryan Friedman <ryanfriedman5410+github@gmail.com>
Date:   Sun Mar 17 16:51:53 2024 -0600

    Dev: Add ROS 2 ethernet support information
    
    Signed-off-by: Ryan Friedman <ryanfriedman5410+github@gmail.com>

dev/source/docs/ros.rst
dev/source/docs/ros2-over-ethernet.rst

commit cf7cc27682044b56eac6d8bc2baaa7094a09806d
Author: Ryan Friedman <ryanfriedman5410+github@gmail.com>
Date:   Sun Mar 17 16:31:08 2024 -0600

    README: Add missing newline
    
    Signed-off-by: Ryan Friedman <ryanfriedman5410+github@gmail.com>

README.md
```